qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/common/data_handling/test/test_data_presenter.py

```diff
@@ -291,6 +291,82 @@ def test_handle_table_cell_changed_checkbox_ticked(self):
         self.presenter.plot_added_notifier.notify_subscribers.assert_any_call(self.ws1)
         self.assertEqual(0, self.presenter.plot_removed_notifier.notify_subscribers.call_count)
 
+    @patch(dir_path + ".data_presenter.ADS")
+    def test_handle_table_cell_changed_ticked_to_plot_on_dspacing_when_tof_plotted(self, ads_patch):
+        self.view.get_item_checked.side_effect = _get_item_checked_mock
+        self.presenter.row_numbers = data_presenter.TwoWayRowDict()
+        self.presenter.row_numbers["TOF_WS_Name1"] = 0
+        self.presenter.row_numbers["dSpacing_WS_Name1"] = 1
+
+        ws2 = mock.MagicMock()
+        ws2.getXDimension().name = "dSpacing"
+        self.model.get_active_ws.return_value = ws2
+        self.presenter.plot_added_notifier = mock.MagicMock()
+        self.presenter.plot_removed_notifier = mock.MagicMock()
+        self.presenter.plotted.add("TOF_WS_Name1")
+        ads_patch.retrieve().getXDimension().name = "TOF"
+
+        self.presenter._handle_table_cell_changed(1, 2)  # tick the 2nd dSpacing row to be plotted
+
+        self.assertEqual(1, self.view.set_item_checkstate.call_count)
+        self.assertEqual(0, self.presenter.plot_added_notifier.notify_subscribers.call_count)
+        self.assertEqual(1, len(self.presenter.plotted))
+        self.assertEqual("TOF_WS_Name1", next(iter(self.presenter.plotted)))
+        self.view.set_item_checkstate.assert_called_once_with(1, 2, False)
+
+    @patch(dir_path + ".data_presenter.ADS")
+    def test_handle_table_cell_changed_ticked_to_plot_on_tof_when_tof_plotted(self, ads_patch):
+        self.view.get_item_checked.side_effect = _get_item_checked_mock
+        self.presenter.row_numbers = data_presenter.TwoWayRowDict()
+        self.presenter.row_numbers["TOF_WS_Name1"] = 0
+        self.presenter.row_numbers["TOF_WS_Name2"] = 1
```

Comment:
Some of this setup seems to be very similar in a few tests (probably in the existing ones as well!), I wonder if it could be put in a helper function, where you just supply a list of names and xunits you need? What do you think?